src/tests/Loader/ContextualReflection/ContextualReflection.cs

```diff
@@ -729,11 +729,12 @@ void TestActivatorCreateInstance(bool isolated)
         void TestDefineDynamicAssembly(bool collectibleContext, AssemblyBuilderAccess assemblyBuilderAccess)
         {
             AssemblyLoadContext assemblyLoadContext = collectibleContext ? new AssemblyLoadContext("DynamicAssembly Collectable context", true) : AssemblyLoadContext.Default;
-            AssemblyName dynamicAssemblyName = new AssemblyName("DynamicAssembly");
+            AssemblyBuilder assemblyBuilder;
+            AssemblyName dynamicAssemblyName = new AssemblyName($"DynamicAssembly_{Guid.NewGuid():N}");
 
             using (assemblyLoadContext.EnterContextualReflection())
             {
-                AssemblyBuilder assemblyBuilder = AssemblyBuilder.DefineDynamicAssembly(dynamicAssemblyName, assemblyBuilderAccess);
+                assemblyBuilder = AssemblyBuilder.DefineDynamicAssembly(dynamicAssemblyName, assemblyBuilderAccess);
             }
 
             Assert.IsTrue(assemblyLoadContext.Assemblies.Any(a => AssemblyName.ReferenceMatchesDefinition(a.GetName(), dynamicAssemblyName)));
```

Comment:
I think it should make use of the assemblyBuilder outside of the using block - just to be on the safe side. Otherwise the compiler/JIT can throw it away before it gets to the validation.

Reply:
Fixed